### ISSUE TITLE
WA Menu Filters do not function properly after the project is imported (USFM)

### ISSUE THREAD

**Issue by elsylambert** [unfoldingWord/translationCore#7350]
This is a dup of #7349

- Open an aligned Project and then export it to USFM with Alignments.
- In TC, choose the option Import locally to import the USFM file from step 1.
- After the import, open the project>wA tool>Select GWL and Launch
- Click on the Menu filters checkboxes. Note that the filters do not work and do not show any results.

**Comment by PhotoNomad0** [comment]
@elsylambert  What I observe after importing a 100% aligned bible, open it in WA and try the filters:
- invalidated filter on - see no verses  ✅
- bookmark filter on - see no verses  ✅
- completed filter on - see no verses, they will appear as complete is toggled  ✅
- incompleted filter on - see all verses, they will disappear from list as complete is toggled  ✅
- verse edit filter on - see no verses  ✅
- comments filter on - see no verses  ✅
- unaligned words filter on - see no verses  ✅

Remember when we export to USFM we don't preserve any history data - no bookmarks, no completed flags, no verse edit history, no comments, no tN and tW selections.  Only the actual alignments if checkbox is selected.

**Comment by Samuel-Kim** [comment]
1. Imported the project https://git.door43.org/ru_gl/ru_rlob_neh_tcore_checked_tW-checked_tN-checked

- tCore warned: Changes have been detected in your project which have invalidated some of your alignments. These verses display an icon of a broken chain next to the reference in the side menu.
- Word Alignment was 99% complete and contained 2 broken chains.
- translationWords was 100% complete and contained 0 broken chains.

2. selected Invalidated displayed 2 Alignment uncomplete places.
3. straightened broken words.
4. selected Invalid No results
5. selected Bookmark No results
6. selected Completed all verses
7. selected Incomplete No results
8. selected Verse edit
9. selected Comments
10. selected Unaligned words

- 2:13 contains no alignment words
- 5:7 contains no alignment words

Status: - Word Alignment was 100% complete and contained 0 broken chains.

11. I am exporting a project to usfm
12. check the box to include alignment data
13. Import local project from usfm file
14. create a project ru_test_neh_book

- Word Alignment was 98% complete and contained 0 broken chains
- translationWords was 0% complete and contained 0 broken chains

15. Choose Gateway Language: English (en) - Door43-Catalog
16. I press Launch.
17. Gateway Language did not appear in the **Step 1. Read** field. Only Original Language and Target Language were displayed.
18. selected Invalid No results
19. selected Bookmark No results
20. selected Completed No results
21. selected Incomplete all verses
22. selected Verse edit No results
23. selected Comments No results
24. selected Unaligned words

- 8:2
- 12:7
- 12:26
- 12:46
- 12:47

25. lined up the words

Status: - Word Alignment was 100% complete and contained 0 broken chains.

26. I am exporting a project to usfm
27. check the box to include alignment data
28. Import local project from usfm file
29. create a project ru_tes_neh_book

- Word Alignment was 100% complete and contained 0 broken chains
- translationWords was 0% complete and contained 0 broken chains

30. Choose Gateway Language: English (en) - Door43-Catalog
31. I press Launch.
32. Gateway Language did not appear in the **Step 1. Read** field. Only Original Language and Target Language were displayed.
33. selected Invalid No results
34. selected Bookmark No results
35. selected Completed No results
36. selected Incomplete all verses
37. selected Verse edit No results
38. selected Comments No results
39. selected Unaligned words No results

**Comment by Samuel-Kim** [comment]
[Wed Jul 06 2022.log](https://github.com/unfoldingWord/translationCore/files/9061510/Wed.Jul.06.2022.log)

**Comment by PhotoNomad0** [comment]
>     10. selected Unaligned words
> 
> 
>     * 2:13 contains no alignment words
> 
>     * 5:7 contains no alignment words
> 
> 

@Samuel-Kim - Thanks for the write up.  We have been splitting up each problem we find in your videos into separate issues so that they can be easily tracked and worked on. The item 10 is written up in https://github.com/unfoldingword/translationcore/issues/7349.  You can see a list of all the issues created attached to the Epic https://github.com/unfoldingWord/translationCore/issues/7342.  It is still a work in process with @elsylambert and I working on it.

**Comment by PhotoNomad0** [comment]
@elsylambert Addressing the issue that when we import aligned bibles from USFM we do not see the completed verses in the completed filter.  This is due to a past decision that we do not want to consider alignments for a verse complete until a human has reviewed it - either by manual toggling the alignment complete button or automatically when the user has aligned all the words by dragging.

@birchamp Do you want to discuss with the tools team to see if we should change this default behavior?

**Comment by birchamp** [comment]
@birchamp discuss with content team.

**Comment by PhotoNomad0** [comment]
Notes:

Retesting this with latest develop build:

> 1. Imported the project https://git.door43.org/ru_gl/ru_rlob_neh_tcore_checked_tW-checked_tN-checked
> 
> 
>     * tCore warned: Changes have been detected in your project which have invalidated some of your alignments. These verses display an icon of a broken chain next to the reference in the side menu.
> 
>     * Word Alignment was 99% complete and contained 2 broken chains.
> 
>     * translationWords was 100% complete and contained 0 broken chains.
> 
> 
>     2. selected Invalidated displayed 2 Alignment uncomplete places.
> 
>     3. straightened broken words.
> 
>     4. selected Invalid No results
> 
>     5. selected Bookmark No results
> 
>     6. selected Completed all verses
> 
>     7. selected Incomplete No results
> 
>     8. selected Verse edit
> 
>     9. selected Comments
> 
>     10. selected Unaligned words
> 
> 
>     * 2:13 contains no alignment words
> 
>     * 5:7 contains no alignment words
> 
> 
> Status: - Word Alignment was 100% complete and contained 0 broken chains.
> 
>     11. I am exporting a project to usfm
> 
>     12. check the box to include alignment data
> 
>     13. Import local project from usfm file
> 
>     14. create a project ru_test_neh_book
> 
> 
>     * Word Alignment was 98% complete and contained 0 broken chains

with latest develop it now shows 100% complete
> 
>     * translationWords was 0% complete and contained 0 broken chains
> 
> 
>     15. Choose Gateway Language: English (en) - Door43-Catalog

with latest develop it now shows 100% complete
> 
>     16. I press Launch.
> 
>     17. Gateway Language did not appear in the **Step 1. Read** field. Only Original Language and Target Language were displayed.
> 
>     18. selected Invalid No results
> 
>     19. selected Bookmark No results
> 
>     20. selected Completed No results
> 
>     21. selected Incomplete all verses
> 
>     22. selected Verse edit No results
> 
>     23. selected Comments No results
> 
>     24. selected Unaligned words

with latest develop it now shows no unaligned words
> 
> 
>     * 8:2
> 
>     * 12:7
> 
>     * 12:26
> 
>     * 12:46
> 
>     * 12:47
> 
> 
>     25. lined up the words
> 
> 
> Status: - Word Alignment was 100% complete and contained 0 broken chains.
> 
>     26. I am exporting a project to usfm
> 
>     27. check the box to include alignment data
> 
>     28. Import local project from usfm file
> 
>     29. create a project ru_tes_neh_book
> 
> 
>     * Word Alignment was 100% complete and contained 0 broken chains
> 
>     * translationWords was 0% complete and contained 0 broken chains
> 
> 
>     30. Choose Gateway Language: English (en) - Door43-Catalog
> 
>     31. I press Launch.
> 
>     32. Gateway Language did not appear in the **Step 1. Read** field. Only Original Language and Target Language were displayed.
> 
>     33. selected Invalid No results
> 
>     34. selected Bookmark No results
> 
>     35. selected Completed No results
> 
>     36. selected Incomplete all verses
> 
>     37. selected Verse edit No results
> 
>     38. selected Comments No results
> 
>     39. selected Unaligned words No results

**Comment by PhotoNomad0** [comment]
@elsylambert This should be fixed in translationCore 3.4.1 (56de0cc).  See notes above ^^^ .  It looks like The fixes made to migrate alignments to latest original language have also cleaned up the imports.

**Comment by elsylambert** [comment]
Verified in translationCore 3.4.1 (06cf68c). I exported to USFM a fully aligned project  with alignments and imported it back from the local USFM. I see all the alignments are there but wA filter does not show the results.
<img width="1642" alt="Screenshot 2023-01-13 at 9 52 18 AM" src="https://user-images.githubusercontent.com/51806180/212349282-13324483-9ebd-4998-b6da-09ddf8b9ece8.png">
@PhotoNomad0 Am I missing something here?
